Maths/MeanAbsoluteDeviation.js

```diff
@@ -0,0 +1,23 @@
+import { mean } from './AverageMean.js'
+/**
+ *@function meanAbsoluteDeviation
+ *@description Calculates the mean absolute deviation of list of numbers
+ * @param {Integer} data
+ * @returns meanAbsoluteDeviation([2,34,5,0,-2]) = 10.480
+ * @url https://en.wikipedia.org/wiki/Average_absolute_deviation
+ */
+function meanAbsoluteDeviation (data) {
+  if (!Array.isArray(data)) {
+    throw new TypeError('Invalid Input')
+  }
+  let absoluteSum = 0
+  const meanValue = mean(data)
+  for (const dataPoint of data) {
+    absoluteSum += Math.abs(dataPoint - meanValue)
+  }
+  return parseFloat((absoluteSum / data.length).toFixed(3))
```

Comment:
Why the `parseFloat`, and why truncate to the first 3 digits?

Reply:
I thought that much accuracy is not required. Therefore truncated to the first 3 decimals.

Reply:
But how does truncating help with that? Truncation may be appropriate for tests, but not for the implementation / algorithm.

Reply:
Right. I have updated them.